help_to_heat/frontdoor/interface.py

```diff
@@ -162,37 +169,52 @@ class SupplierConverter:
     def __init__(self, session_id):
         self.session_id = session_id
 
-    def _get_supplier(self):
-        return api.session.get_answer(self.session_id, "supplier")["supplier"]
+    def get_supplier(self):
+        supplier = api.session.get_answer(self.session_id, supplier_page).get(supplier_field)
+        alternative_supplier = api.session.get_answer(self.session_id, alternative_supplier_page).get(
+            alternative_supplier_field
+        )
+
+        return alternative_supplier if self._did_specify_alternative() else supplier
```

Comment:
I don't think there's real overhead to `api.session.get_answer` is there? We could be quite a bit more efficient with how many times we're calling it if so.

Reply:
don't think so too much, though it is a database call. have reworked it to just check once